bindings/core/src/method_handler/secret_manager.rs

```diff
@@ -19,12 +19,13 @@ pub(crate) async fn call_secret_manager_method_internal(
 ) -> Result<Response> {
     let response = match method {
         SecretManagerMethod::GenerateAddresses { options } => {
-            let addresses = GetAddressesBuilder::new(secret_manager)
-                .set_options(options)?
-                .finish()
-                .await?;
+            let addresses = secret_manager.get_addresses(options).await?;
```

Comment:
I'm a bit triggered by the mix of `generate_addresses` and `get_addresses`. I'd say it's more generating than getting, maybe we can harmonise.

Reply:
Fair point.